### ISSUE TITLE
copy_to fails on column; works on all others

### ISSUE THREAD

**Issue by csheehey** [rstudio/sparklyr#637]
Hi,

I'm attempting to load a local data frame into Spark, but receive an error due to a seemingly innocuous field.  Here is the reproducible example and error codes:

```r

library(sparklyr)
library(dplyr)
library(httr)

sc <- spark_connect(master = "local", version = "1.6.0")

# DOWNLOAD DATA from data.gov (CFPB Customer Complaint database)
# assign api endpoint
url <- "https://data.consumerfinance.gov/resource/jhzv-w97w.csv"
limit = 5  # number of records to retrieve

# get data
response <- httr::GET(paste0(url, "?$limit=", as.character(limit)))

# convert to dataframe
complaint_df <- data.frame(content(response, as = "parsed"))

# transfer to spark
copy_to(sc, complaint_df, overwrite = TRUE)
### RECEIVES ERROR...

# remove 5th column: "complaint_what_happened"
complaint_df_strip <- complaint_df %>% select(-5)

# transfer to spark
copy_to(sc, complaint_df_strip, overwrite = TRUE)
### VOILA.  Load completes successfully

```

And the error message (abbreviated):

```

Error: org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 31.0 failed 4  times, most recent failure: Lost task 0.3 in stage 31.0 (TID 36, <master alias>): java.lang.ArrayIndexOutOfBoundsException: 5

```

**Comment by edgararuiz** [comment]
@csheehey , what's the 5th field's name? Does it have any special characters?

**Comment by csheehey** [comment]
Hi @edgararuiz 

The field name is "complaint_what_happened".  The CFPB masks all sensitive information in the field, replacing it with characters of the type "XX/XX/XXXX".  Here's an example from the third row... maybe the newline character at the end?

"I have been struggling with my mortgage payments, and applied for loan modification, back in XX/XX/XXXX.I was sent a package to resubmit back to mortgage holder, providing current information on income, expenses and desired options. The mortgage holder keeps on requesting the same income information that was previously provided and insist on delaying procedure to underwriter. HUD Counselor has take the approach that there is no standard or guideline to delays and stall tactics being used in the decision process rather a wait and see.I think from XX/XX/XXXX to XX/XX/XXXX is more than enough time to give a reasonable respond for an application. \n"

**Comment by edgararuiz** [comment]
that's a possibility, can we try removing newline in `complaint_df` before shipping it to spark?

**Comment by csheehey** [comment]
Voila!  That did the trick...

```

complaint_df_fix <- complaint_df %>% 
  mutate(complaint_what_happened_fix = str_replace_all(complaint_what_happened, "[\r\n]" , "")) %>%
  select(-complaint_what_happened)

copy_to(sc, complaint_df_fix, overwrite = TRUE)

Source:   query [5 x 18]
Database: spark connection master=yarn-client app=sparklyr local=FALSE

                            company                          company_public_response        company_response complaint_id consumer_consent_provided
                              <chr>                                            <chr>                   <chr>        <int>                     <chr>
1                   Bank of America Company chooses not to provide a public response Closed with explanation      1780421                       N/A
2               Synchrony Financial                                               NA Closed with explanation      1543488          Consent provided
3                             Ocwen                                               NA Closed with explanation      1820018          Consent provided
4 Fifth Third Financial Corporation                                               NA Closed with explanation      2019326                       N/A
5              SunTrust Banks, Inc.                                               NA Closed with explanation       629839                       N/A
# ... with 13 more variables: consumer_disputed <chr>, date_received <dbl>, date_sent_to_company <dbl>, issue <chr>, product <chr>, state <chr>,
#   sub_issue <chr>, sub_product <chr>, submitted_via <chr>, tags <chr>, timely <chr>, zip_code <chr>, complaint_what_happened_fix <chr>

```

Big thanks @edgararuiz !
